addons/mail/static/src/utils/utils.js

```diff
@@ -13,6 +13,35 @@ const {
 
 const classPatchMap = new WeakMap();
 
+/**
+ * Returns whether the given event has been handled by the given handlerName.
+ *
+ * @param {Event} ev
+ * @param {string} handlerName
+ * @returns {boolean}
+ */
+function isEventHandledBy(ev, handlerName) {
+    if (!ev.__oHandledBy) {
+        return false;
+    }
+    return ev.__oHandledBy.includes(handlerName);
+}
+
+/**
+ * Marks the given event as handled by the given handlerName. Useful to allow
+ * handlers in the propagation chain to make a decision based on what has
+ * already been done.
+ *
+ * @param {Event} ev
+ * @param {string} handlerName
+ */
+function markEventHandledBy(ev, handlerName) {
+    if (!ev.__oHandledBy) {
+        ev.__oHandledBy = [];
```

Comment:
Note: if this is considering too optimistic or not clean, it should possible to achieve the same with a WeakMap.

Reply:
done